feature-teacher/outing/src/commonMain/kotlin/com.b1nd.dodam.outing/OutScreen.kt

```diff
@@ -0,0 +1,498 @@
+                                            .background(
+                                                shimmerEffect(),
+                                                RoundedCornerShape(50.dp),
+                                            ),
+                                    )
+                                    Box(
+                                        modifier = Modifier
+                                            .size(width = 71.dp, height = 27.dp)
+                                            .padding(start = 8.dp)
+                                            .background(
+                                                shimmerEffect(),
+                                                RoundedCornerShape(8.dp),
+                                            ),
+                                    )
+                                    Spacer(modifier = Modifier.weight(1f))
+
+                                    Box(
+                                        modifier = Modifier
+                                            .size(width = 71.dp, height = 27.dp)
+                                            .background(
+                                                shimmerEffect(),
+                                                RoundedCornerShape(8.dp),
+                                            ),
+                                    )
+                                }
+                            }
+                        }
+                    }
+
+                    is OutPendingUiState.Success -> {
+                        val cnt =
+                            if (titleIndex == 0) data.outPendingCount else data.sleepoverPendingCount
+                        val members =
+                            if (titleIndex == 0) data.outMembers else data.sleepoverMembers
+                        var filteredMemberList = if (gradeIndex == 0 && roomIndex == 0) {
+                            members
+                        } else if (gradeIndex == 0 && roomIndex != 0) {
+                            members.filter { studentData ->
+                                studentData.student.room == roomIndex
+                            }
+                        } else if (gradeIndex != 0 && roomIndex == 0) {
+                            members.filter { studentData ->
+                                studentData.student.grade == gradeIndex
+                            }
+                        } else {
+                            members.filter { studentData ->
+                                studentData.student.grade == gradeIndex && studentData.student.room == roomIndex
+                            }
+                        }
+                        if (searchStudent.isNotEmpty()) {
+                            filteredMemberList = filteredMemberList.filter {
+                                it.student.name.contains(searchStudent) == true
+                            }
+                        }
```

Comment:
```suggestion
                        val members = if (titleIndex == 0) data.outMembers else data.sleepoverMembers
                        val filteredMemberList = members.filter { studentData ->
                            when {
                                gradeIndex == 0 && roomIndex == 0 -> true
                                gradeIndex == 0 && roomIndex != 0 -> studentData.student.room == roomIndex
                                gradeIndex != 0 && roomIndex == 0 -> studentData.student.grade == gradeIndex
                                else -> studentData.student.grade == gradeIndex && studentData.student.room == roomIndex
                            }
                        }.let { filteredList ->
                            if (searchStudent.isNotEmpty()) {
                                filteredList.filter { it.student.name.contains(searchStudent) }
                            } else {
                                filteredList
                            }
                        }
```
와 같이 개선하면 좋을 것 같습니다.